### PR TITLE
Fix Jakarta Data checkpoint tests for DB Rotation

### PR DESCRIPTION
**Pull request by KyleAure**
Fixes defect where the following errors occurs during server startup: 

```txt
java.lang.RuntimeException: java.sql.SQLNonTransientException: DSRA4000E: No implementations of org.postgresql.xa.PGXADataSource are found for dataSource[DefaultDataSource] with library JDBCLibrary. The name or location of JDBC driver JAR files might be incorrect or inaccessible. Searched in: [/home/ci/Build/workspace/ebcTestRunner/dev/autoFVT/image/output/wlp/usr/shared/resources/jdbc/derby.jar]. Searched under packages: [org.postgresql.xa].
```

The server.xml needs to be updated to use the JDBC Driver for the Database being used during the build.
Also added the extra permissions to avoid future failures.

### CODE REVIEW

**Review by njr-11 — DISMISSED** [review]

**Comment by KyleAure** [comment]
#build

**Comment by KyleAure** [comment]
RTC Build: Passed
Pipeline: Failed - checkpoint tests still failed.  I'm guessing the reason is because environment variables are not persisted between server restarts.  Going to debug.

**Comment by KyleAure** [comment]
#libby - recheck headers
#build - redo personal build

**Comment by KyleAure** [comment]
Forgot to add a blank server.env file.  Rebuilding
#libby - check headers
#build - rebuild

**Comment by KyleAure** [comment]
#libby - recheck headers
#build - personal build

**Comment by KyleAure** [comment]
#libby - check headers
#build - redo personal build

**Comment by KyleAure** [comment]
#libby - recheck headers
#build - personal build

**Comment by KyleAure** [comment]
#libby - recheck headers
#build - personal build

**Comment by KyleAure** [comment]
Custom Database SOE: https://libh-proxy1.fyre.ibm.com/cognitive-dev/pipelineAnalysis.html?pipelineId=81dcf816-eaa9-4bea-a712-17c00619cf13

Target locations of links might be accessible only to IBM employees.

**Comment by KyleAure** [comment]
Pipeline Build: Failed with two known defects
RTC Build: Failed three tests due to fyre networking issues
DB Rotation SOE: Passed, but did not run checkpoint tests

Once reviewed this PR is good to be merged.

File: dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/DataExtensionProvider.java
Comment (njr-11) [inline dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/DataExtensionProvider.java:249]: This appears to be okay because CheckpointPhase.onRestore is documented as running the operation immediately if not using a checkpoint/restore.  This is very confusing and when we remove the TODO in a future PR we should add some explanation that it will do this so future maintainers can understand this code.

**Review by njr-11 — APPROVED** [review]

**Comment by KyleAure** [comment]
#libby - re-check headers
#build - re-do personal build after resolving merge conflict

**Comment by KyleAure** [comment]
Pipeline build: Failed with two known defects
RTC build: Failed with two known defects

File: dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/DataExtensionProvider.java
Comment (tjwatson) [reply dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/DataExtensionProvider.java:249]: I'm open to another name for `onRestore` if it can make things more clear.  I didn't really like `runNowOrOnRestore`.

Comment (njr-11) [reply dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/DataExtensionProvider.java:249]: I don't have very good ideas either - `CheckpointPhase.onRestoreElseRun(...)`.  We will either need to find a name that makes it clear that it runs the code even if not using checkpoint/restore, or otherwise ensure we include comments where it is used. The existing name is too misleading and makes it look like the supplied code is only for the restore path.

**Pull request merged**
